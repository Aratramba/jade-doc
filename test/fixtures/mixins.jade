//- @jadedoc
  name: mixin1

mixin mixin1
  | this is a mixin


//- @jadedoc
  name: mixin2
  description: this is jade mixin 2 documentation
  arguments: 
    arg1: foo
    arg2: faa

mixin mixin2(arg1, arg2)
  div this is a mixin #{arg1} #{arg2}