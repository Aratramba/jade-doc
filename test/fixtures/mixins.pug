//- @pugdoc
  name: mixin
  description: this is jade mixin documentation
  arguments: 
    - {string} arg1 - this is arg1
    - {string} arg2 - this is arg2
  attributes:
    - {string} attr1 - this is attr1
    - {string} attr2 - this is attr2
    - {string} data-attr3 - this is attr3
  examples:
    - +mixin2('foo', 'faa') // mixin example description
    - +mixin2('faa', 'foo')
    -
      name: Mixin 2c
      examples:
        - |
          +mixin2('beep', 'boop')

mixin mixin2(arg1, arg2)
  div this is a mixin #{arg1}
  div this is the same mixin #{arg2}