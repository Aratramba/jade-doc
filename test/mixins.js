'use strict';
/* globals require */

var rimraf = require('rimraf');
rimraf.sync('./test/tmp');

var JadeDoc = require('../index');
new JadeDoc({
  input: './test/fixtures/**/*.jade',
  output: './test/tmp/all.json'
});