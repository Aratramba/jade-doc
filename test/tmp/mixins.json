[{"meta":{"name":"mixin","description":"this is jade mixin documentation","arguments":{"arg1":"foo","arg2":"faa"}},"file":"test/fixtures/mixins.jade","source":"mixin mixin2(arg1, arg2)\n  div this is a mixin #{arg1}\n  div this is the same mixin #{arg2}","output":"<div>this is a mixin foo</div><div>this is the same mixin faa</div>"}]