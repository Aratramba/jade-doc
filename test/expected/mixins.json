{
  "mixin1": {
    "meta": {
      "file": "./test/fixtures/mixins.jade"
    },
    "jade": {
      "root": "mixin mixin1",
      "comment": "  name: mixin1",
      "block": "mixin mixin1\n  | this is a mixin",
      "type": "mixin"
    },
    "name": "mixin1",
    "output": "this is a mixin"
  },
  "mixin2": {
    "meta": {
      "file": "./test/fixtures/mixins.jade"
    },
    "jade": {
      "root": "mixin mixin2(arg1, arg2)",
      "comment": "  name: mixin2\n  description: this is jade mixin 2 documentation\n  arguments: \n    arg1: foo\n    arg2: faa",
      "block": "mixin mixin2(arg1, arg2)\n  div this is a mixin #{arg1} #{arg2}",
      "type": "mixin"
    },
    "name": "mixin2",
    "description": "this is jade mixin 2 documentation",
    "arguments": {
      "arg1": "foo",
      "arg2": "faa"
    },
    "output": "<div>this is a mixin foo faa</div>"
  }
}